tasks/install_powershell.ps1

```diff
@@ -6,10 +6,19 @@ Param(
 # If an error is encountered, the script will stop instead of the default of "Continue"
 $ErrorActionPreference = "Stop"
 
-if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {
-    $arch = "x86"
-} else {
+try {
+  if ((Get-WmiObject Win32_OperatingSystem).OSArchitecture -match '^32') {
+      $arch = "x86"
+  } else {
+      $arch = "x64"
+  }
+}
+catch [System.Management.Automation.CommandNotFoundException] {
```

Comment:
When I was looking in to adding integration tests for puppet5/6. 

I was trying to run against our vagrant setup (mwrock/WindowsNano). I started debugging the install_powershell.ps1 puppet_agent::install implementation. I was getting the deprecation error for using `GetWmiObject` (note that part of this work was adding `bolt` user as local admin in the vagrant config).

I also had to use an alternate download method `Invoke-WebRequest`  to get the MSI because `system.net.webclient` was not available (that did not seem worth adding  at the time though I guess it could given that we have `wget`, `curl`, etc in the shell implementation). 

Once I added those workarounds I got to the `msiexec` step and found that windows installer (MSI) is not supported for windows nano (https://docs.microsoft.com/en-us/windows-server/get-started/deploy-nano-server). So I decided that I should focus on testing windows install at the acceptance level.